### PR TITLE
✨feat: Enhance support for containers

### PR DESCRIPTION
**Pull request by alperencelik**
This PR aims to enhance LXC support for Proxmox. Here is the documentation of LXC(Containers) API.

https://pve.proxmox.com/pve-docs/api-viewer/index.html#/nodes/{node}/lxc

### CODE REVIEW

**Comment by alperencelik** [comment]
I'm not sure if this PR satisfies #118.

File: containers.go [hunk 1]
```diff
@@ -52,24 +52,28 @@ func (c *Container) Start(ctx context.Context) (status string, err error) {
 	return status, c.client.Post(ctx, fmt.Sprintf("/nodes/%s/lxc/%d/status/start", c.Node, c.VMID), nil, &status)
 }
 
-func (c *Container) Stop(ctx context.Context) (status *ContainerStatus, err error) {
+func (c *Container) Stop(ctx context.Context) (status string, err error) {
```
Comment (luthermonson) [inline containers.go:55]: oh did we really add this before the the data field was being auto-parsed... goodness that was a long time ago. great fix and probably can remove the ContainerStatus type too right? it's at this point useless

File: containers.go [hunk 2]
```diff
@@ -52,24 +52,28 @@ func (c *Container) Start(ctx context.Context) (status string, err error) {
...
+func (c *Container) Resume(ctx context.Context) (status string, err error) {
 	return status, c.client.Post(ctx, fmt.Sprintf("/nodes/%s/lxc/%d/status/resume", c.Node, c.VMID), nil, &status)
 }
 
+func (c *Container) Shutdown(ctx context.Context, force bool, timeout int) (status string, err error) {
+	return status, c.client.Post(ctx, fmt.Sprintf("/nodes/%s/lxc/%d/status/shutdown", c.Node, c.VMID), map[string]interface{}{"force": force, "timeout": timeout}, &status)
+}
+
 func (c *Container) TermProxy(ctx context.Context) (vnc *VNC, err error) {
-	return vnc, c.client.Post(ctx, fmt.Sprintf("/nodes/%s/lxk/%d/termproxy", c.Node, c.VMID), nil, &vnc)
```
Comment (luthermonson) [inline containers.go:72]: 🙈

File: containers.go [hunk 3]
```diff
@@ -78,3 +82,176 @@ func (c *Container) VNCWebSocket(vnc *VNC) (chan string, chan string, chan error
...
+
+func (c *Container) MoveVolume(ctx context.Context, params *VirtualMachineMoveDiskOptions) (task *Task, err error) {
+	var upid UPID
+	if err := c.client.Post(ctx, fmt.Sprintf("/nodes/%s/lxc/%d/move_volume", c.Node, c.VMID), params, &upid); err != nil {
+		return nil, err
+	}
+	return NewTask(upid, c.client), nil
+}
+
+func (c *Container) RRDData(ctx context.Context, timeframe Timeframe, consolidationFunction ConsolidationFunction) (rrddata []*RRDData, err error) {
```
Comment (luthermonson) [inline containers.go:125]: any chance you learned anything writing the container RRDData func to help fix the virtualmachine func for this guy https://github.com/luthermonson/go-proxmox/issues/112 ?

**Comment by luthermonson** [comment]
@alperencelik this pretty much what most people need to get going with containers. Closes #118 imho

If you think we can drop that type and/or know of a fix for the rrddata func for vms then feel free to drop another PR. going to get this merged and cut a release tonight as this is a ton of new funcs added that people may way to use

File: containers.go (hunk 3 above)
Comment (alperencelik) [reply containers.go:125]: Well, I didn't realize the issue before so I just followed the same method for Containers as we do on VirtualMachines.  If you think that can cause issues we can remove or I can try to follow up on the issue and if I find anything I can come back to this one.

File: containers.go (hunk 1 above)
Comment (alperencelik) [reply containers.go:55]: Actually when I checked the API doc realized it turns the UPID as a string so I don't think we need something like containerStatus. IMO returning the UPID as it is might not be the correct way to do it so we can convert them to tasks. Here is an example of start but any status command returns the same object:

`root@lowtower:~# pvesh create /nodes/lowtower/lxc/150/status/start
UPID:lowtower:0035154F:068C3DB0:65E99E9D:vzstart:150:root@pam:`

So what I'm offering is converting that one to 

`func (c *Container) Stop(ctx context.Context) (task *Task, err error) {
	var upid UPID
	if err := c.client.Post(ctx, fmt.Sprintf("/nodes/%s/lxc/%d/status/stop", c.Node, c.VMID), nil, &upid); err != nil {
		return nil, err
	}
	return NewTask(upid, c.client), nil
}`

**Comment by alperencelik** [comment]
Thank you so much for reviewing @luthermonson. 

I would like to hear your thoughts as well so for /lxc/status I'm offering to convert the strings to tasks since the /status returns a UPID but for /rrdData I guess I need some time to test it.

File: containers.go (hunk 1 above)
Comment (luthermonson) [reply containers.go:55]: i agree to that change, let's do that. UPID and Tasks didnt exist when those funcs were initially wrote so let's go that route.

File: containers.go (hunk 3 above)
Comment (luthermonson) [reply containers.go:125]: i have no context, i thought it worked tbh and havent looked into it

**Pull request merged**